### PR TITLE
Feature/cam 24332 merge bluetooth status manager library into beacons library

### PR DESCRIPTION
**Pull request by andywEXL**

### CODE REVIEW

File: android/src/main/java/de/patwoz/rn/bluetoothstatemanager/RNBluetoothStateManagerModule.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+  }
+
+  @ReactMethod
+  public void addListener(String eventName) {}
+
+  @ReactMethod
+  public void removeListeners(Integer count) {}
+
+  @ReactMethod
+  public void disable(Promise promise) {
```
Comment (BeezyWorks) [inline android/src/main/java/de/patwoz/rn/bluetoothstatemanager/RNBluetoothStateManagerModule.java:124]: looks like the is and enable are basically the same function. maybe we should extract the shared bits

File: android/src/main/java/de/patwoz/rn/bluetoothstatemanager/RNBluetoothStateManagerModule.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+      return true;
+    }
+    return false;
+  }
+
+  private boolean handleIfBluetoothNotSupported(Promise promise)  {
+    return this.handleIfBluetoothNotSupported(promise, true);
+  }
+
+  private Activity handleCurrentActivity(Promise promise) {
```
Comment (BeezyWorks) [inline android/src/main/java/de/patwoz/rn/bluetoothstatemanager/RNBluetoothStateManagerModule.java:249]: why is this called handleCurrentActivity, should it be get?

File: ios/RNiBeacon/RNBTStateManager/RNBluetoothStateManager.m [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+// ----------------------------------------------------------------------------------------------- -
+// NOT AVAILABLE IN iOS
+
+RCT_EXPORT_METHOD(enable:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+  NSError* error = nil;
+  reject(@"UNSUPPORTED", @"Not implemented in iOS", error);
+}
+
+RCT_EXPORT_METHOD(disable:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
```
Comment (BeezyWorks) [inline ios/RNiBeacon/RNBTStateManager/RNBluetoothStateManager.m:77]: if we're simply going to reject the promises for these functions, maybe better not to have them at all. No sense having an API split by platform on our cross platform lib

**Review by BeezyWorks — APPROVED** [review]

File: android/src/main/java/de/patwoz/rn/bluetoothstatemanager/RNBluetoothStateManagerModule.java (hunk 1 above)
Comment (DuduExlibris) [reply android/src/main/java/de/patwoz/rn/bluetoothstatemanager/RNBluetoothStateManagerModule.java:124]: Why do we introducing the ebable/disable methods? shouldn't it be just about getting the state?

**Comment by andywEXL** [comment]
We are going to keep the original react-native-bluetooth-state-manager code as is to start with.  There may be relevant uses for enable in our product (if user has turned off bluetooth but needs it.
In general we can allocate a ticket to clean things up here going forward.

**Pull request merged**